pkg/registry/apis/iam/team_search.go

```diff
@@ -0,0 +1,167 @@
+package iam
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"slices"
+	"strconv"
+
+	"go.opentelemetry.io/otel/trace"
+	common "k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/spec3"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+
+	iamv0alpha1 "github.com/grafana/grafana/apps/iam/pkg/apis/iam/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/apiserver/builder"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	teamsearch "github.com/grafana/grafana/pkg/services/team/search"
+	"github.com/grafana/grafana/pkg/storage/legacysql/dualwrite"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+	"github.com/grafana/grafana/pkg/storage/unified/resourcepb"
+	"github.com/grafana/grafana/pkg/util/errhttp"
+)
+
+type TeamSearchHandler struct {
+	log      log.Logger
+	client   resourcepb.ResourceIndexClient
+	tracer   trace.Tracer
+	features featuremgmt.FeatureToggles
+}
+
+func NewTeamSearchHandler(tracer trace.Tracer, dual dualwrite.Service, legacyTeamSearcher resourcepb.ResourceIndexClient, resourceClient resource.ResourceClient, features featuremgmt.FeatureToggles) *TeamSearchHandler {
+	searchClient := resource.NewSearchClient(dualwrite.NewSearchAdapter(dual), iamv0alpha1.TeamResourceInfo.GroupResource(), resourceClient, legacyTeamSearcher, features)
+
+	return &TeamSearchHandler{
+		client:   searchClient,
+		log:      log.New("grafana-apiserver.teams.search"),
+		tracer:   tracer,
+		features: features,
+	}
+}
+
+func (s *TeamSearchHandler) GetAPIRoutes(defs map[string]common.OpenAPIDefinition) *builder.APIRoutes {
```

Comment:
Almost the same question as the previous: can we use the app platform sdk to generate this? (custom routes)

@IfSentient Does the codegen support generating the models and the OpenAPI defs?